docs/uploading_and_publishing/explanation_how_the_processing_works.md

```diff
@@ -0,0 +1,122 @@
+
+In the previous section, [How-to upload data](howto_upload_data.md), we showed that when the data was dropped in the Uploads page, a processing of the raw data was triggered.
+
+<div class="click-zoom">
+    <label>
+        <input type="checkbox">
+        <img src="/assets/uploading_and_publishing/gwupload.gif" alt="GW upload gif" width="90%" title="Click to zoom in">
+    </label>
+</div>
+
+
+Once the data is added to the upload, NOMAD interprets the files and divides them into two categories: **mainfiles** and **auxiliary files**. In the same upload, there might be multiple mainfiles and auxiliary files organized in a folder tree structure.
+
+The **mainfiles** are those files which are representative of a given computational calculation. The presence of a mainfile in the upload is key for NOMAD to recognize a calculation. In NOMAD, we support several computational codes for first principles calculations, molecular dynamics simulations, and lattice modeling, as well as workflow and database managers. For each code, NOMAD recognizes a single file as the mainfile. For example, the [VASP](https://www.vasp.at/) mainfile is by default the `vasprun.xml`, although if the `vasprun.xml` is not present in the upload NOMAD searches the `OUTCAR` file and assigns it as the mainfile[^1]. 
```

Comment:
We should connect mainfile with the entries in nomad.

Reply:
Entries are explained a bit down, on Structured data with the NOMAD metainfo.